### PR TITLE
[Console] cursor tweaks

### PR DESCRIPTION
**Pull request by fabpot**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no 
| Tickets       | Refs #27444
| License       | MIT
| Doc PR        | n/a

While playing with the new cursor class, I realized that I would like to make some changes. Mainly having a fluent interface: `$cursor->moveUp(10)->clearOutput()` for instance.

I have also separater the `clearLine()` method in two method to avoid the bool arg.

/cc @pierredup

### CODE REVIEW

File: src/Symfony/Component/Console/Cursor.php [hunk 1]
```diff
@@ -28,82 +27,114 @@ public function __construct(OutputInterface $output, $input = STDIN)
         $this->input = $input;
     }
 
-    public function moveUp(int $lines = 1)
+    public function moveUp(int $lines = 1): self
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Console/Cursor.php:30]: since the class is not final (should it be?), we need to add `@return $this` anotations

Comment (fabpot) [reply src/Symfony/Component/Console/Cursor.php:30]: Class is now final :)

File: src/Symfony/Component/Console/Cursor.php [hunk 2]
```diff
@@ -28,82 +27,114 @@ public function __construct(OutputInterface $output, $input = STDIN)
...
-        }
+        $this->output->write("\x1b[2K");
+
+        return $this;
+    }
+
+    /**
+     * Clears all the output from the current line after the current position.
+     */
+    public function clearLineAfter(): self
```
Comment (pierredup) [inline src/Symfony/Component/Console/Cursor.php:113]: [QuestionHelper.php#L356](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Console/Helper/QuestionHelper.php#L356) would also need to be updated to use this new method

Comment (fabpot) [reply src/Symfony/Component/Console/Cursor.php:113]: Indeed, fixed.

**Review by pierredup — APPROVED** [review]

**Review by chalasr — APPROVED** [review]

**Review by nicolas-grekas — APPROVED** [review]

**Pull request merged**